clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/LeaderFollowerStoreIngestionTask.java

```diff
@@ -1960,64 +1960,72 @@ protected boolean isHybridFollower(PartitionConsumptionState partitionConsumptio
     return isHybridMode() && (isDaVinciClient || partitionConsumptionState.getLeaderFollowerState().equals(STANDBY));
   }
 
+  /**
+   * For non AA hybrid stores with AGGREGATE DRP, SIT reads from parent RT while the HB is written to the child RTs.
+   * Once all hybrid stores are either AA for cross colo replication and non AA otherwise, DRP and this extra
+   * check can also be removed.
+   */
+  protected boolean shouldCheckLeaderCompleteStateInFollower() {
+    return (getServerConfig().isLeaderCompleteStateCheckInFollowerEnabled() && this.hybridStoreConfig.isPresent()
+        && this.hybridStoreConfig.get().getDataReplicationPolicy() != DataReplicationPolicy.AGGREGATE);
+  }
+
   /**
    * Checks whether the lag is acceptable for hybrid stores
    * <p>
-   * leaderCompleteStateCheckEnabled is not used for non AA stores: as SIT reads from parent RT for non AA
-   * stores but HB is written to local RT. This should be enabled once all hybrid stores are made AA.
+   * If the instance is a hybrid standby or DaVinciClient: Also check if <br>
```

Comment:
Can we also add comment that leader completed check is done only if lag has caught up against VT?